From 04983e06138607fb6557ac9895873fed813b3d58 Mon Sep 17 00:00:00 2001
From: shuvokr <anik@sdlknvadijfasdjn.local>
Date: Wed, 10 Nov 2021 23:02:32 +0600
Subject: [PATCH 21/29] - Added product info view model action.

---
 .../ModelView/ProductModelView.swift          | 21 ++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/Asimple_e-market_application/ModelView/ProductModelView.swift b/Asimple_e-market_application/ModelView/ProductModelView.swift
index 3c263e2..04d5717 100644
--- a/Asimple_e-market_application/ModelView/ProductModelView.swift
+++ b/Asimple_e-market_application/ModelView/ProductModelView.swift
@@ -8,5 +8,24 @@
 import UIKit
 
 class ProductModelView: NSObject {
-
+    private var apiService : APIService!
+    private(set) var storeData : [Products]! {
+        didSet {
+            self.bindProductInfoViewModelToController()
+        }
+    }
+    
+    var bindProductInfoViewModelToController : (() -> ()) = {}
+    
+    override init() {
+        super.init()
+        self.apiService =  APIService()
+        callFuncToGetProductData()
+    }
+    
+    func callFuncToGetProductData() {
+        self.apiService.apiToGetProductsInfo { (storeData) in
+            self.storeData = storeData
+        }
+    }
 }
-- 
2.30.1 (Apple Git-130)

